pkg/verifier/types/types.go

```diff
@@ -79,6 +83,30 @@ func WriteVerifyResultResult(result *verifier.VerifierResult, w io.Writer) error
 	return json.NewEncoder(w).Encode(result)
 }
 
+// CreateVerifierResult creates a new verifier result object from given input.
+func CreateVerifierResult(verifierName, verifierType, message string, isSuccess bool, err *errors.Error) VerifierResult {
```

Comment:
Can you clarify what is the difference between `VerifierResult` struct here and `VerifierResult` struct in pkg/verifier/result.go?

Reply:
Sure, I have to admit it's confusing between the 2 similar verifierResult.
Firstly, let's look at structure of them, the left one is under pkg/verifier/types/types.go, the right one is under pkg/verifier/result.go
![image](https://github.com/user-attachments/assets/e5c1327b-be6b-47f6-b95e-23d3b681ba1f)

We can see the right one contains all fields in left one and additional fields like `Subject`, `ReferenceDigest` and `ErrorReason` etc.

Now back to the functionality and usage of `verifierReports`. The left(short) one is the direct result generated by a verifier plugin. So it will only contain the minimal fields from verifier validation. But the right(long) one is the result that's populated when generating the verification response to Gatekeeper, so it adds additional info that's out of scope of the verifier itself.

Reply:
ahh I see thanks for the explanation.